src/robot/parsing/model/statements.py

```diff
@@ -968,15 +968,20 @@ class WhileHeader(Statement):
     type = Token.WHILE
 
     @classmethod
-    def from_params(cls, condition, limit=None, indent=FOUR_SPACES,
-                    separator=FOUR_SPACES, eol=EOL):
+    def from_params(cls, condition, limit=None, on_limit_message=None,
+                    indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
         tokens = [Token(Token.SEPARATOR, indent),
                   Token(cls.type),
                   Token(Token.SEPARATOR, separator),
                   Token(Token.ARGUMENT, condition)]
         if limit:
             tokens.extend([Token(Token.SEPARATOR, indent),
                            Token(Token.OPTION, f'limit={limit}')])
+        if on_limit_message:
+            tokens.extend([Token(Token.SEPARATOR, indent),
+                           Token(Token.OPTION,
+                                 f'on_limit_message={on_limit_message}'
+                                 )])
```

Comment:
Closing parens should be at the end of the previous line for consistency with other code.